app/src/main/java/com/sopt/now/data/ApiFactory.kt

```diff
@@ -0,0 +1,29 @@
+package com.sopt.now.data
+
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import com.sopt.now.BuildConfig
+import com.sopt.now.data.data.AuthService
+import com.sopt.now.data.data.InfoService
+import com.sopt.now.data.data.LoginService
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType.Companion.toMediaType
+import retrofit2.Retrofit
+
+object ApiFactory {
+    private const val BASE_URL = BuildConfig.AUTH_BASE_URL
+
+    val retrofit: Retrofit by lazy {
+        Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
+            .build()
+    }
+
+    inline fun <reified T> create(): T = retrofit.create(T::class.java)
+}
+
+object ServicePool {
+    val authService = ApiFactory.create<AuthService>()
+    val loginService = ApiFactory.create<LoginService>()
+    val infoService = ApiFactory.create<InfoService>()
+}
```

Comment:
```suggestion
object ServicePool {
    val authService: AuthService by lazy { create<AuthService>() }
    val loginService: AuthService by lazy { create<LoginService>() }
    val infoService: InfoService by lazy { create<InfoService>() }
}
```

이러면 서비스 중 하나만 사용되어도 ServicePool 객체가 참조될 때 나머지 서비스들도 모두 초기화되는 문제가 있습니다! 불필요한 리소스를 유발 할 것 같아요.

특히 네트워크 관련 서비스의 경우 초기화 과정에서 추가적인 시간이 소요될 수 있으므로 비효율적 일 수도 있을 것 같습니다!

Reply:
생각 못하고 있던 부분이었는데 감사합니다!